lib/generator.js

```diff
@@ -692,6 +698,38 @@ class Generator {
 
     return Promise.all(promises);
   }
+
+  /**
+   * Check if given directory is a git repo with unstaged changes or is not empty
+   * @private
+   * @param  {String} [dir] Directory that needs to be tested for a given condition.
```

Comment:
I think `dir` is not optional, so we should remove the square brackets:

```suggestion
   * @param  {String} dir Directory that needs to be tested for a given condition.
```